File: workbook-6/demos/01-interfaces-intro-final/src/main/java/com/pluralsight/models/RidingMower.java
package com.pluralsight.models;

import com.pluralsight.models.interfaces.PowerTool;

public class RidingMower extends Vehicle implements PowerTool
{
    public RidingMower(String make, String model)
    {
        super(make, model);
    }

    @Override
    public void repair()
    {
        System.out.println("repairing RidingMower ... overrides Vehicle and PowerTool contracts");
    }
}

File: workbook-6/demos/02-review-python-final/riding_mower.py
from vehicle import Vehicle
from power_tool import PowerTool

class RidingMower(Vehicle,PowerTool):
    
    def __init__(self, make, model):
        Vehicle.__init__(self, make, model)
        PowerTool.__init__(self)
        print("Bulding a new Riding Mower")
        
    def repair(self):
        Vehicle.repair(self)
        PowerTool.repair(self)
